### PR TITLE
Tgl h ipc4 toml

### PR DESCRIPTION
**Pull request by juimonen**

### CODE REVIEW

**Review by lyakh — APPROVED** [review]

**Comment by juimonen** [comment]
needed: https://github.com/thesofproject/sof/pull/5946

**Review by kv2019i — CHANGES_REQUESTED** [review]
You have the openssl3.0 guards in the same PR. Plus some notes on the tgl-h specifics.

File: config/tgl-h-cavs.toml [hunk 1]
```diff
@@ -0,0 +1,215 @@
+version = [2, 5]
+
+[adsp]
+name = "tgl"
+image_size = "0x1F0000"
```
[the remark this answers is not in the document]
Comment (juimonen) [reply config/tgl-h-cavs.toml:5]: ok

**Review by kv2019i — APPROVED** [review]
Thanks, looks fine now!

**Review by fredoh9 — APPROVED** [review]

**Comment by marc-hb** [comment]
> I do see additional change on src/pkcs1_5.c. 

My favorite git topic! :-)

That's because github does not support `git range-diff`. To see the real difference you can fetch and run `git range-diff` locally, see demo below.

The other, better workaround is to NOT rebase on force-push. Unless there are conflicts rebasing is pointless and noisy because Github and CI do a merge anyway. You can see that merge with `git fetch origin pull/98/merge`.

```shell

git fetch origin 80b497f5abd933932456e22775740ad61c5ed661
git tag _sof98_1 80b497f5abd933932456e22775740ad61c5ed661

git fetch origin 570fd341cee1208e58c9d499abcf3d62673864e0
git tag _sof98_2 570fd341cee1208e58c9d499abcf3d62673864e0

git range-diff _sof98_1..._sof98_2
```
```diff
1:  73a9d7c897d3 < -:  ------------ rimage: fix openssl 3.0 support in ver25 signing
2:  5e599a23ac62 < -:  ------------ rimage: ace signing functions need openssl 3.0 guards
3:  80b497f5abd9 ! 1:  570fd341cee1 toml: add tgl-h config file for IPC4
    @@ config/tgl-h-cavs.toml (new)
     +
     +[adsp]
     +name = "tgl"
    -+image_size = "0x1F0000"
    ++image_size = "0x1F0000" # (30 + 1) bank * 64KB
     +
     +[[adsp.mem_zone]]
     +type = "ROM"
```

Gerrit implemented `git range-diff` years ago, see demo at: https://chromium-review.googlesource.com/c/chromiumos/third_party/kernel/+/346001/25..26/drivers/gpu/drm/i915/intel_display.c

Gitlab did not yet https://gitlab.com/gitlab-org/gitlab/-/issues/24096

File: config/tgl-h-cavs.toml (hunk 1 above)
[the remark this answers is not in the document]
Comment (kv2019i) [reply config/tgl-h-cavs.toml]: @fredoh9  @marc-hb This is correct now. The image size should equal sum of HP and LP SRAM. tgl has more (46+1)*64kB==0x2f0000, while tgl-h has less (30+1)*64kB==0x1f0000

[the remark this answers is not in the document]
Comment (marc-hb) [reply config/tgl-h-cavs.toml]: > The image size should equal sum of HP and LP SRAM

Does this needs a comment or is this obvious enough? It didn't seem obvious for a couple other people besides me...

**Review by lgirdwood — APPROVED** [review]

**Comment by lgirdwood** [comment]
@juimonen I assume you can send another PR for rimage to sof ?

**Pull request merged**